glue/flagging_script_glue/flagging.py

```diff
@@ -767,114 +767,156 @@ def z_normalize_groupby(s: pd.Series):
+            df["sv_short_owner"] == "Short-term owner",
+            df["sv_name_match"] != "No match",
+            df[["sv_buyer_category", "sv_seller_category"]]
+            .eq("legal_entity")
+            .any(axis=1),
+            df["sv_anomaly"] == "Outlier",
+            df["sv_pricing"].str.contains("High price swing")
+            | df["sv_pricing"].str.contains("Low price swing"),
         ]
 
-        labels = [
-            "Home flip sale (high)",
-            "Family sale (high)",
-            "Non-person sale (high)",
-            "Anomaly (high)",
-            "High price swing",
-            "High price (raw)",
-            "Home flip sale (low)",
-            "Family sale (low)",
-            "Non-person sale (low)",
-            "Anomaly (low)",
-            "Low price swing",
-            "Low price (raw)",
+        # Define labels for characteristic-based reasons
+        char_labels = [
+            "Short-term owner",
+            "Family sale",
+            "Non-person sale",
+            "Statistical anomaly",
+            "Price Swing / Home flip",
         ]
 
+        # Define conditions for price-based reasons
+        price_conditions = [
+            df["sv_pricing"].str.contains("High"),
+            df["sv_pricing"].str.contains("Low"),
+        ]
+
+        # Define labels for price-based reasons
+        price_labels = [
+            "High price",
+            "Low price",
+        ]
+
+        # Apply np.select to create the new columns
+        df["sv_char_outlier_reason"] = np.select(
+            char_conditions, char_labels, default="Not outlier"
+        )
+        df["sv_char_price_reason"] = np.select(
+            price_conditions, price_labels, default="Not outlier"
+        )
+
     else:
-        conditions = [
-            (df["sv_short_owner"] == "Short-term owner")
-            & (df["sv_pricing"].str.contains("High")),
-            (df["sv_name_match"] != "No match")
-            & (df["sv_pricing"].str.contains("High")),
-            (
-                df[["sv_buyer_category", "sv_seller_category"]]
-                .eq("legal_entity")
-                .any(axis=1)
-            )
-            & (df["sv_pricing"].str.contains("High")),
-            (df["sv_anomaly"] == "Outlier") & (df["sv_pricing"].str.contains("High")),
-            (df["sv_pricing"].str.contains("High price swing")),
+        # Define conditions for characteristic-based reasons
+        char_conditions = [
+            df["sv_short_owner"] == "Short-term owner",
+            df["sv_name_match"] != "No match",
+            df[["sv_buyer_category", "sv_seller_category"]]
+            .eq("legal_entity")
+            .any(axis=1),
+            df["sv_anomaly"] == "Outlier",
+            df["sv_pricing"].str.contains("High price swing"),
+            df["sv_anomaly"] == "Outlier",
+            df["sv_pricing"].str.contains("High price swing")
+            | df["sv_pricing"].str.contains("Low price swing"),
+        ]
```

Comment:
**[Question, non-blocking]** It seems like res and condos share the same set of `char_conditions`, is that right? If so, perhaps we could define one `char_conditions` list to be shared by both workflows, and then have `price_conditions` be distinct depending on the value of `condos`?